### PR TITLE
fixed joins association

### PR DESCRIPTION
**Pull request by stasl**

### CODE REVIEW

**Comment by jonkessler** [comment]
Thanks for the contribution! Would you mind adding a test for this change?

**Comment by stasl** [comment]
@jonkessler I have added spec and few more fixes, please check

File: lib/store_base_sti_class_for_4_1.rb [hunk 1]
```diff
@@ -65,7 +65,12 @@ def join_constraints(foreign_table, foreign_klass, node, join_type, tables, scop
               end
               scope_chain_index += 1
 
-              scope_chain_items.concat [klass.send(:build_default_scope, ActiveRecord::Relation.create(klass, table))].compact
+              default_scope = if klass.method(:build_default_scope).arity == 0
+                klass.send(:build_default_scope)
+              else
+                klass.send(:build_default_scope, ActiveRecord::Relation.create(klass, table))
+              end
+              scope_chain_items.concat [default_scope].compact
```
Comment (jonkessler) [inline lib/store_base_sti_class_for_4_1.rb]: Is this change in Rails 4.1.x somewhere?

Comment (stasl) [inline lib/store_base_sti_class_for_4_1.rb]: yes, method signature was change in 4.1.x, then added default parameter.
specs failed for that versions,  this is how I discovered

Comment (jonkessler) [inline lib/store_base_sti_class_for_4_1.rb]: Awesome. Thanks!

File: lib/store_base_sti_class_for_4_2.rb [hunk 2]
```diff
@@ -231,6 +299,27 @@ def construct_join_attributes(*records)
 
       end
 
+      class HasManyThroughAssociation
+
+        def build_through_record(record)
```
Comment (jonkessler) [inline lib/store_base_sti_class_for_4_2.rb]: Everything looks great. Would you mind adding one more test for this change?

Comment (stasl) [inline lib/store_base_sti_class_for_4_2.rb]: spec I added cover this case already

Comment (jonkessler) [inline lib/store_base_sti_class_for_4_2.rb]: Ah, you're right! Sorry.

**Comment by jonkessler** [comment]
Thanks!

**Pull request merged**